### DISCUSSION TITLE
Argo rollouts canary trafficRouting with Istio Ambient mode using HTTPRoute instead of VIrtualService

### DISCUSSION THREAD

**Discussion by proxyvert** [argoproj/argo-rollouts#3897] [General]
Hello Argo Rollouts community,

I'm trying to implement canary deployments using Argo Rollouts in an environment with the following setup:

- Istio Ambient mode (https://istio.io/latest/docs/ambient/getting-started/deploy-sample-app/)
- Kubernetes Gateway API (using Gateway and HTTPRoute resources)
- Argo Rollouts for managing deployments

My goal is to use Argo Rollouts to manage canary deployments while leveraging the Kubernetes Gateway API for traffic routing. However, I'm unclear on how to properly configure this integration.

Specific questions:

1. Does Argo Rollouts currently support direct integration with Kubernetes Gateway API resources (specifically HTTPRoute) for traffic splitting in canary deployments?

2. If not, is there a recommended approach or best practice for using Argo Rollouts in conjunction with Kubernetes Gateway API and Istio Ambient mode?

3. Are there plans to add native support for Kubernetes Gateway API in future releases of Argo Rollouts?

4. In the meantime, what would be the most effective way to implement canary deployments in this environment? Should we consider using Istio VirtualServices instead, or implement a custom controller to update HTTPRoute resources based on Rollout progress?

Any guidance or best practices would be greatly appreciated. Thank you for your time and assistance!

Environment details:
- Server Version: 4.14.0-0.okd-2024-01-06-084517 (we run OKD)
  Kubernetes version: v1.27.1-3474+5c56cc35e3edc1-dirty

- Istio version: 1.23.2
- Argo Rollouts version: v1.7.2+59e5bd3
- HTTPRoute api version: gateway.networking.k8s.io/v1

**Comment by kostis-codefresh** [comment]
>Does Argo Rollouts currently support direct integration with Kubernetes Gateway API resources (specifically HTTPRoute) for traffic splitting in canary deployments?

Yes :-)
https://rollouts-plugin-trafficrouter-gatewayapi.readthedocs.io/en/latest/
